elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchFlowStage.scala

```diff
@@ -23,33 +23,58 @@ import akka.stream.alpakka.elasticsearch.scaladsl.ElasticsearchSinkSettings
 import org.apache.http.message.BasicHeader
 import org.apache.http.util.EntityUtils
 
-final case class IncomingMessage[T](id: Option[String], source: T)
+object IncomingMessage {
```

Comment:
It might be useful if it also has factory methods to create an instance without id for Java.
```scala
def create[T](source: T): IncomingMessage[T, NotUsed] = IncomingMessage(None, source)
def create[T, C](source: T, cargo: C): IncomingMessage[T, C] =IncomingMessage(None, source, cargo)
```

Reply:
done